src/query/service/src/pipelines/processors/transforms/window/partition/transform_window_partition_collect.rs

```diff
@@ -109,9 +115,24 @@ impl TransformWindowPartitionCollect {
             partition_id[*partition] = new_partition_id;
         }
 
+        let spill_config = SpillerConfig {
+            location_prefix: query_spill_prefix(ctx.get_tenant().tenant_name(), &ctx.get_id()),
+            disk_spill,
+            spiller_type: SpillerType::Window,
+        };
+
+        // Create an inner `Spiller` to spill data.
+        let operator = DataOperator::instance().operator();
+        let spiller = Spiller::create(ctx, operator, spill_config)?;
+
         // Create the window partition buffer.
+        let sort_block_size = settings.get_window_partition_sort_block_size()? as usize;
         let buffer =
-            WindowPartitionBuffer::new(ctx, partitions.len(), sort_block_size, spill_settings)?;
+            WindowPartitionBuffer::new(spiller, num_partitions, sort_block_size, spill_settings)?;
```

Comment:
Use `partitions.len()` instead of `num_partitions` here, because `num_partitions` is the total number of partitions, and `partitions.len()` is the number of partitions that this processor needs to handle.